tests/test_userspace.py

```diff
@@ -87,3 +91,64 @@ def test_hostname(shell, check):
         assert serial_number in etc_hostname
 
 
+@pytest.fixture
+def clocktree(shell):
+    """
+    Read the clock tree from the DUT and parse it into a data structure.
+    """
+    re_entry = re.compile(r"^\s*(\S+)\s+\d+\s+\d+\s+\d+\s+(\d+)\s+\d+\s+\d+\s+(\d+)\s+\S\s+(\S+)\s+")
+    re_2nd = re.compile(r"^\s+(\S+)\s+\S+\s+$")
```

Comment:
The `\s\S\s\s\S\S…` sure looks like [Parseltongue](https://en.wikipedia.org/wiki/Magic_in_Harry_Potter#Parseltongue) to me, but I assume it works.